keys.go

```diff
@@ -132,9 +133,12 @@ func FetchKeysDirect(serverName ServerName, addr, sni string) (*ServerKeys, *tls
 	if err != nil {
 		return nil, nil, err
 	}
+	if response.StatusCode != http.StatusOK {
+		return nil, nil, fmt.Errorf("Remote host responded with non-OK status when attempting to fetch server keys: %s", response.Status)
```

Comment:
I think this could be clearer/more succinct:

```suggestion
		return nil, nil, fmt.Errorf("Non-200 response %d from remote server", response.StatusCode)
```